lib/chef/handler/datadog.rb

```diff
@@ -118,6 +119,87 @@ def restore_env_proxies
         ENV['http_proxy'] = @env_http_proxy
         ENV['https_proxy'] = @env_https_proxy
       end
+
+      # create and configure all the Dogapi Clients to be used
+      #
+      # @return [Array] all Dogapi::Client to be used
+      def prepare_the_pack
+        dogs = []
+        endpoints.each do |url, api_key, app_key|
+          dogs.push(Dogapi::Client.new(
+                      api_key,
+                      app_key,
+                      nil,   # host
+                      nil,   # device
+                      true,  # silent
+                      nil,   # timeout
+                      url
+          ))
+        end
+        dogs
+      end
+
+      def convert_to_array(value_or_array)
+        return value_or_array if value_or_array.nil?
+        value_or_array.is_a?(Array) ? value_or_array : [value_or_array]
+      end
+
+      # return all endpoints as a list of triplets [url, api_key, application_key]
+      def endpoints
+        urls = convert_to_array @config[:url]
+        api_keys = convert_to_array @config[:api_key]
+        app_keys = convert_to_array @config[:application_key]
+
+        validate_extra_endpoints(urls, api_keys, app_keys)
+
+        endpoints = []
+        if urls.nil? || urls.length == 1
+          url = urls.nil? ? nil : urls[0]
+          [api_keys.length, app_keys.length].min.times do |index|
+            endpoints.push [url, api_keys[index], app_keys[index]]
+          end
+        else
+          [urls.length, api_keys.length, app_keys.length].min.times do |index|
+            endpoints.push [urls[index], api_keys[index], app_keys[index]]
+          end
+        end
+        endpoints
+      end
+
+      # Validate endpoints config (urls, api_keys and application keys)
+      # fails if incorrect and nothing can be done
+      # Doesn't fail if at least one endpoint can be used
+      def validate_extra_endpoints(urls, api_keys, app_keys)
```

Comment:
naming nitpick: I'd prefer `validate_endpoints` since it doesn't apply only to the extra endpoints